src/api/app/models/bs_request/reviews_overview.rb

```diff
@@ -0,0 +1,72 @@
+# This class is only responsible to show overview of the reviews and remove
+# duplication from `requested by` list
+class BsRequest::ReviewsOverview
```

Comment:
Hm I thought we do this some when else, but let me repeat what I wrote on the [card in the backlog](https://trello.com/c/NysbjAZj).

Those really sound like scopes on `Review` to me. Something like...

```
scope :by_project, -> { where(by_package: nil).where.not(by_project: nil).where(project: { staging_workflow: nil } )
scope :staged, -> { where(by_package: nil).where.not(by_project: nil).where.not(project: { staging_workflow: nil } )
scope :by_package, -> { where.not(by_package: nil).where.not(by_project: nil) }
scope :by_user, -> { where.not(by_user: nil) }
scope :by_group, -> { where.not(by_group: nil) }
```

The only special handling you will have to do in the view then is mimic the order of your current hash.

Please note there are, confusingly, some finder scopes with the same names in `BsRequest`.

Reply:
It doesn't sound like a query object or a scope anymore. There is a lot of custom logic to remove duplicate reviews. For [example](https://github.com/openSUSE/open-build-service/pull/12863/files#diff-8626573888e5460dd0d35672519418fa9072cbbdee75421f98deb1b76b40fac9R52) shouldn't be a scope and this is not a query as well. What do you think?

Reply:
Why do you remove duplicated reviews?

Reply:
That's something I suggested because we had something like this:
![Screenshot 2022-08-09 at 13-21-44 Open Build Service](https://user-images.githubusercontent.com/2581944/183635640-264fdb07-aca8-4de2-9cbc-cfb3303136ae.png)

The same text three times, with no difference between them (no clue about the "Comment for reviewer").
That's why I thought it was enough to simply represent something like "Admin, you have something to review" once.

 But of course, if removing the duplication makes the code a lot more complex, let's ignore my suggestion and display the same thing three times.

Reply:
Yea there is a reason we do not disallow this. Reviews (for the same person) have state and can happen at various times during of the request. You should not remove them.

Reply:
This is already addressed